pkg/controllers/workspace_controller.go

```diff
@@ -359,6 +403,41 @@ Retry_withdifferentname:
 	return resources.GetNode(ctx, newMachine.Status.NodeName, c.Client)
 }
 
+func (c *WorkspaceReconciler) CreateNodeClaim(ctx context.Context, wObj *kaitov1alpha1.Workspace, nodeOSDiskSize string) (*corev1.Node, error) {
+RetryWithDifferentName:
+	newNodeClaim := nodeclaim.GenerateNodeClaimManifest(ctx, nodeOSDiskSize, wObj)
+
+	if err := nodeclaim.CreateNodeClaim(ctx, newNodeClaim, c.Client); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			klog.InfoS("There exists a nodeClaim with the same name, retry with a different name", "nodeClaim", klog.KObj(newNodeClaim))
+			goto RetryWithDifferentName
```

Comment:
Consider for loop...to avoid infinite loop and better code readability

const maxRetries = 5
for i := 0; i < maxRetries; i++ {
newNodeClaim := nodeclaim.GenerateNodeClaimManifest(ctx, nodeOSDiskSize, wObj)

       if err := nodeclaim.CreateNodeClaim(ctx, newNodeClaim, c.Client); err != nil {
            if apierrors.IsAlreadyExists(err) {
                klog.InfoS("There exists a nodeClaim with the same name, retrying with a different name", "nodeClaim", klog.KObj(newNodeClaim))
                continue
            }
            ....

Reply:
But we don't want to quit until we find a new name.